### DISCUSSION TITLE
removed

### DISCUSSION THREAD

**Discussion by shells-above** [QIN2DIM/hcaptcha-challenger#59] [Q&A]

**Comment by QIN2DIM** [comment]
resnet？ yolo？

**Reply by QIN2DIM** [reply]
I recommend using yolov5n6, where speed of solution is more important on such tasks.

**Reply by QIN2DIM** [reply]
I support your attempt to do so. We have initially practiced this approach (pluggable ONNX models) in hcaptcha challenger. We split the original model, which can do several classifications, into binary models with the corresponding number of labels, and each ONNX model handles only one individual task.

**Comment by QIN2DIM** [comment]
The visual challenges of hcaptcha and recaptcha are different.

**Comment by yyyanbj** [comment]
I think it's enough for classification, but not great enough for detection task.

**Reply by yyyanbj** [reply]
I think you just need to collect the images which recaptcha used. If you use 8k images, it will cause revolution problems which you may need to use something like multi-scale technology to solve this. It’s really not worth to do this.
I only work for the new challenge, and all of this can only use the classification model to solve. It’s easy to manually label them. Actually, any task is a binary classification, that only true or false.

**Reply by QIN2DIM** [reply]
In reCAPTCHA, you need to deal with "4x4 object detection problem" and "3x3 image classification problem". In business logic, you need to determine whether the current challenge is a classification problem or a detection problem. 

The labels involved in reCAPTCHA are already present in the COCO namespace (like traffic light). You don't need to make a separate dataset for this, it's all open source. On the other hand, I encourage you to get the dataset directly from reCAPTCHA.

- [OpenImages Dataset v6 - traffic light](https://storage.googleapis.com/openimages/web/visualizer/index.html?set=train&type=segmentation&r=false&c=%2Fm%2F015qff)

**Reply by QIN2DIM** [reply]
Today's sophisticated technology in the CV field can completely crush this simple cat-and-mouse game, so I hope you know that the difficulty of reCAPTCHA is not in the visual challenge itself.

**Reply by QIN2DIM** [reply]
Up to you. ପ( ˘ᵕ˘ ) ੭ ☆

This is obviously a demand-driven problem, on some situations, where the weight of accuracy is much lower than passing, I was able to choose a more lightweight, easily deployable, and portable solution.

It's an art of balance ᐕ)ﾉ
